BurnerB/FreeMentors-V2
# [ #167962035] User signup API

#### What does this PR do?
Allows a user to signup
#### Description of Task to be completed?
* Create user signup model and controller
* Create user input validations
* Tests should pass
* Refractor tests
#### How should this be manually tested?
* Clone this  repository https://github.com/BurnerB/FreeMentors-V2.git and cd into FreeMentors-V2
* checkout to branch ft-User-signup-api-167962035
*Run npm --init to install all dependancies.
* Run command npm test on your terminal
* Tests should pass
* Open postman and POST `/api/v1/auth/signup`
* Enter the spec:
 ```
    "firstName":"John",
    "lastName":"Doe",
	"email":"johndoe@email.com",
	"password":"password123",
	"address":"Nairobi",
    "bio":"rapper and record producer known as one of the best artists",
    "occupation":"Musician",
    "expertise":"rapping"
 ```
 * Should recieve response:
 ```
 "status": 201,
    "message": "User created successfully",
    "data": {
        "token": "eyJhbGciOiJIUzI1NiIsI......"
      }
 ```
#### What are the relevant pivotal tracker stories?
[#167962035](https://www.pivotaltracker.com/story/show/167962035)
#### Screenshots (if appropriate) 

![success_signup](https://user-images.githubusercontent.com/45785786/63327032-1f29e380-c336-11e9-85fc-77ff7d99bd78.png)
